Anlab.Mvc/Services/DocumentSigningService.cs

```diff
@@ -0,0 +1,229 @@
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Net.Http;
+using System.Threading.Tasks;
+using Anlab.Core.Data;
+using Anlab.Core.Domain;
+using AnlabMvc.Models.Configuration;
+using DocuSign.eSign.Api;
+using DocuSign.eSign.Client;
+using DocuSign.eSign.Client.Auth;
+using DocuSign.eSign.Model;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Options;
+
+namespace AnlabMvc.Services;
+
+public interface IDocumentSigningService
+{
+    Task<string> SendForEmbeddedSigning(int orderId);
+    Task<FileStreamResult> DownloadEnvelope(string envelopeId);
+    OAuth.UserInfo GetAccountInfo(OAuth.OAuthToken authToken);
+    OAuth.OAuthToken AuthenticateWithJwt();
+    string BuildConsentUrl();
+}
+
+public class DocumentSigningService : IDocumentSigningService
+{
+    private readonly ApplicationDbContext _dbContext;
+    private readonly IHttpContextAccessor _httpContextAccessor;
+    private readonly ESignatureOptions _eSignatureSettings;
+
+    private static readonly string SignerClientId = "1000";
+
+    public DocumentSigningService(IOptions<ESignatureOptions> eSignatureSettings, ApplicationDbContext dbContext, IHttpContextAccessor httpContextAccessor)
+    {
+        _dbContext = dbContext;
+        _httpContextAccessor = httpContextAccessor;
+        _eSignatureSettings = eSignatureSettings.Value;
+    }
+
+    public async Task<string> SendForEmbeddedSigning(int orderId)
+    {
+        var order = await _dbContext.Orders.Include(o=>o.Creator).SingleAsync(a => a.Id == orderId);
+
+        if (order == null)
+        {
+            throw new ArgumentException("Order not found");
+        }
+
+        // first, get auth token
+        var authToken = AuthenticateWithJwt();
+
+        var user = GetAccountInfo(authToken);
+
+        var account = user.Accounts[0]; // TODO: what if there are multiple accounts?  need to select one somehow perhaps

```

Comment:
@srkirkland 
Get all user accounts and look for is_default?
https://developers.docusign.com/platform/auth/reference/user-info/

Reply:
Yes that seems like a good thing to do and the right solution.